packages/api/cms-api/src/dam/dam.module.ts

```diff
@@ -29,48 +29,30 @@ import { DamScopeInterface } from "./types";
 interface DamModuleOptions {
     damConfig: DamConfig;
     imgproxyConfig: ImgproxyConfig;
-}
-
-interface DamModuleAsyncOptions extends Pick<ModuleMetadata, "imports"> {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    useFactory: (...args: any[]) => Promise<DamModuleOptions> | DamModuleOptions;
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    inject?: any[];
+    Scope?: Type<DamScopeInterface>;
 }
 
 @Global()
 @Module({})
 export class DamModule {
-    static registerAsync(options: DamModuleAsyncOptions, Scope?: Type<DamScopeInterface>): DynamicModule {
-        const optionsProvider = {
-            provide: DAM_MODULE_OPTIONS,
-            ...options,
-        };
-
-        const damConfigProvider = {
+    static register({ damConfig, imgproxyConfig, Scope }: DamModuleOptions): DynamicModule {
```

Comment:
:+1: much better! (we should review all modules if they can be simplified)

...and how to upgrade?

Reply:
> we should review all modules if they can be simplified

I believe that most of our modules can be simplified similarly. Most modules are only async to support injecting the config, for instance `KubernetesModule`, `PublicUploadModule`, `BlobStorageModule`.

> ...and how to upgrade?

Maybe we could add this to the v4 release and write a upgrade script? We could also change the DamModule registration in v4